asab/storage/elasticsearch.py

```diff
@@ -468,49 +615,78 @@ async def _insert_noobjid(self):
 
 		# This is insert of the new document, the ObjId is to be generated by the ElasicSearch
 		for url in self.Storage.ServerUrls:
-			url = "{}{}/_doc?refresh={}".format(
+			request_url = "{}{}/_doc?refresh={}".format(
 				url, self.Collection, self.Storage.Refresh
 			)
 
 			try:
-				async with self.Storage.session().request(method="POST", url=url, json=setobj) as resp:
-					if resp.status != 201:
-						raise RuntimeError("Unexpected response code: {}".format(resp.status))
-
-					resp_json = await resp.json()
-					self.ObjId = resp_json['_id']
-					return self.ObjId
+				async with self.Storage.session().request(
+					method="POST",
+					url=request_url,
+					headers=self.Headers,
+					json=upsert_data,
+					ssl=self.SSLcontext
+				) as resp:
+					if resp.status == 401:
+						raise ConnectionRefusedError("Response code 401: Unauthorized. Provide authorization by specifying either user name and password or api key.")
+					elif resp.status not in {200, 201}:
+						raise ConnectionError("Unexpected response code: {}".format(resp.status))
+					else:
+						resp_json = await resp.json()
+						self.ObjId = resp_json['_id']
+						await self.Storage.session().close()
+						return self.ObjId
 
 			except aiohttp.client_exceptions.ClientConnectorError:
 				if url == self.Storage.ServerUrls[-1]:
 					raise Exception("Failed to connect to '{}'".format(url))
 				else:
 					L.warning("Failed to connect to '{}', iterating to another cluster node".format(url))
 
+			except aiohttp.client_exceptions.ServerDisconnectedError:
+				raise Exception("Failed to connect to '{}'".format(url))
+
+			except ValueError as err:
+				raise ConnectionError("Both username and API key specified. Please choose one option. {}".format(err))
```

Comment:
Should it be specifically a ConnectionError? If so, do we want to consider the possibility of changing the comment "Both username and API key specified..."?